addons/website/views/website_views.xml

```diff
@@ -99,7 +99,7 @@
                         <group>
                             <group>
                                 <field name="name"/>
-                                <field name="url"/>
+                                <field name="url" readonly="page_id or is_mega_menu or child_id"/>
```

Comment:
for new_window I would add condition based on is_mega_menu and child_id only, I will keep new_window editable in case of page_id, because in case of page_id url will be readonly but that url can still be opened in new window.